### ISSUE TITLE
clang17 checking not covered

### ISSUE THREAD

**Issue by yannrichet** [r-hub/rhub#564]
Dear Rhub,
I recently updated one of my CRAN packages and received the warning:
```
[...]
The auto-check found additional issues for the *last* version released on CRAN:
  clang17 <https://www.stats.ox.ac.uk/pub/bdr/clang17/rlibkriging.log>
  clang17 <https://www.stats.ox.ac.uk/pub/bdr/clang17/rlibkriging.out>
[...]
```
So I tried to reproduce the problem with rhub/debian-clang-devel , but without sucess.
It seems that beyond **clang**, it also misses **flang** to reveal my portability issue.
So I had to slightly modify debian-clang-devel with following conf :

```
# Install llvm17
apt-get update
apt-get install lsb-release wget software-properties-common gnupg
wget https://apt.llvm.org/llvm.sh
chmod +x llvm.sh
./llvm.sh 17
apt-get install flang-17 

# Setup Makevars to match clang CRAN test
mkdir -p ~/.R
touch ~/.R/Makevars
echo "CC=clang-17" >> ~/.R/Makevars
echo "CXX=clang++-17" >> ~/.R/Makevars
echo "FC=flang-new-17" >> ~/.R/Makevars
```

I don't know if the purpose of this image is really to cover the clang17 CRAN checking, but at least this could help people like me to reproduce the check warning.

Regards.

**Comment by gaborcsardi** [comment]
https://github.com/r-hub/rhub2 has a clang17 image.

**Comment by PascalKieslich** [comment]
Dear R-hub team,

thanks a lot for providing the builder and R package. It has been very helpful for me in testing my packages before submitting to CRAN. 

I just wanted to echo the issue raised by @yannrichet here. It would be super helpful if both clang17 and flang17 were available in the builder as well, as currently these are used in the Debian CRAN pretest. I recently struggled a lot with replicating CRAN check results using R-hub, which, I believe are due to the differences in compilers.

I also tried out https://github.com/r-hub/rhub2, but I believe that changing it / making it available in the builder would make testing easier for the broader R community.

Thanks again, and best regards,

Pascal
